packages/notebook/src/widget.ts

```diff
@@ -234,6 +235,7 @@ export class StaticNotebook extends WindowedList {
     this._updateNotebookConfig();
     this._mimetypeService = options.mimeTypeService;
     this.renderingLayout = options.notebookConfig?.renderingLayout;
+    this.kernelHistory = options.kernelHistory;
```

Comment:
Currently this PR the integration test fails because the factory is still passing `history` rather than `kernelHistory`, see the other two comments.